📖docs/development: Update Mock section (#420)
kubernetes-sigs/cluster-api-provider-azure: docs/development.md

#### Mocks

> CAPZ are using [GoMock][gomock] to generate the services mocks to use in the testing.

Reviewer: ```suggestion
Mocks for the services tests are generated using  [GoMock][gomock]
```

Author: thanks for the review
done.
